src/lobster/mcp/tool_factory.py

```diff
@@ -0,0 +1,105 @@
+"""Tool factory for automatically registering Lobster MCP tools with FastMCP.
+
+This module provides a factory pattern to automatically discover and register
+all available tools with the FastMCP server, eliminating code duplication.
+"""
+
+import inspect
+from typing import Any, get_type_hints
+from collections.abc import Callable
+
+from fastmcp import FastMCP
+
+from .tools import (
+    compute_naturalness,
+    get_sequence_concepts,
+    get_sequence_representations,
+    get_supported_concepts,
+    intervene_on_sequence,
+    list_available_models,
+)
+
+
+class ToolFactory:
+    """Factory for automatically registering Lobster tools with FastMCP."""
+
+    def __init__(self, app: FastMCP):
+        """Initialize the tool factory.
+
+        Parameters
+        ----------
+        app : FastMCP
+            The FastMCP application instance
+        """
+        self.app = app
+        self._registered_tools = {}
+
+    def register_all_tools(self) -> None:
```

Comment:
Your solution is fine and makes a lot of sense, but it does require that new tools are always imported and added here. Another way of doing it would be to move tool registration to where the tools are defined, e.g., you could have a `@register_tool` decorator or something. Or, if new tools in `tools/` are always meant to be registered, you could have some sort of auto-registration that just registers all functions it's able to import from `tools`.